bundles/org.openhab.binding.vallox/src/main/java/org/openhab/binding/vallox/internal/se/handler/ValloxSEHandler.java

```diff
@@ -0,0 +1,479 @@
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Check if reconnection is requested and verify connection is still alive.
+     */
+    private void checkConnection() {
+        if (reconnect) {
+            reconnect = false;
+            closeConnection();
+        }
+        connect();
+    }
+
+    /**
+     * Ensure that all linked channels have been polled at least once and has a value.
+     * OnOffType or DecimalType commands needs a cached value.
+     */
+    private void refreshChannels() {
+        if (isConnected()) {
+            Collection<ChannelDescriptor> channelsToRefresh = linkedChannels().stream().map(ChannelDescriptor::get)
+                    .map(ChannelDescriptor::getParentOrReturn).filter(cache::isExpired).distinct()
+                    .collect(Collectors.toList());
+            logger.debug("Refreshing channels: {}", channelsToRefresh);
+            channelsToRefresh.forEach(this::sendPoll);
+        }
+    }
+
+    /**
+     * CO2 value and set point are 16bit values so high and low bytes are handled separately.
+     *
+     * @param descriptor the descriptor of the channel to be polled
+     */
+    public void sendPoll(ChannelDescriptor descriptor) {
+        if (descriptor.equals(ChannelDescriptor.CO2_SETPOINT)) {
+            sendPollToConnector(ChannelDescriptor.CO2_SETPOINT_HIGH);
+            sendPollToConnector(ChannelDescriptor.CO2_SETPOINT_LOW);
+        } else if (descriptor.equals(ChannelDescriptor.CO2)) {
+            sendPollToConnector(ChannelDescriptor.CO2_HIGH);
+            sendPollToConnector(ChannelDescriptor.CO2_LOW);
+        } else {
+            sendPollToConnector(descriptor);
+        }
+    }
+
+    /**
+     * Forward poll telegram to connection handler.
+     *
+     * @param descriptor the descriptor of the channel to be polled
+     */
+    public void sendPollToConnector(ChannelDescriptor descriptor) {
+        connector.sendTelegram(TelegramFactory.createPoll(panelNumber, descriptor.getVariable()));
+    }
+
+    /**
+     * Forward command telegram to connection handler.
+     *
+     * @param channel the channel which the command is sent to
+     * @param descriptor the descriptor of the channel where the command is sent
+     */
+    public void sendCommand(ChannelDescriptor descriptor, byte value) {
+        connector.sendTelegram(TelegramFactory.createCommand(panelNumber, descriptor.getVariable(), value));
+    }
+
+    /**
+     * Handle telegram received from connection handler.
+     */
+    @Override
+    public void telegramReceived(Telegram telegram) {
+        switch (telegram.state) {
+            case ACK:
+                logger.debug("Received ack byte '{}'", String.format("%02X", telegram.getCheksum()));
+                cache.put(telegram);
+                break;
+            case OK:
+                if (logger.isTraceEnabled()) {
```

Comment:
I think this can be removed? Please see https://community.openhab.org/t/is-checking-logger-istraceenabled-before-logging-a-trace-message-useless/97383/2?u=gitmiguel

Reply:
Done.